presto-parser/src/main/java/com/facebook/presto/sql/tree/FunctionCall.java

```diff
@@ -27,57 +27,55 @@
+        this(Optional.of(location), name, Optional.empty(), Optional.empty(), Optional.empty(), false, arguments);
     }
 
     public FunctionCall(QualifiedName name, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.empty(), name, Optional.empty(), Optional.empty(), distinct, arguments);
-    }
-
-    public FunctionCall(NodeLocation location, QualifiedName name, boolean distinct, List<Expression> arguments)
-    {
-        this(Optional.of(location), name, Optional.empty(), Optional.empty(), distinct, arguments);
+        this(Optional.empty(), name, Optional.empty(), Optional.empty(), Optional.empty(), distinct, arguments);
     }
 
     public FunctionCall(QualifiedName name, Optional<Window> window, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.empty(), name, window, Optional.empty(), distinct, arguments);
+        this(Optional.empty(), name, window, Optional.empty(), Optional.empty(), distinct, arguments);
     }
 
-    public FunctionCall(QualifiedName name, Optional<Window> window, Optional<Expression> filter, boolean distinct, List<Expression> arguments)
+    public FunctionCall(QualifiedName name, Optional<Window> window, Optional<Expression> filter, Optional<List<SortItem>> sortItems, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.empty(), name, window, filter, distinct, arguments);
+        this(Optional.empty(), name, window, filter, sortItems, distinct, arguments);
     }
 
-    public FunctionCall(NodeLocation location, QualifiedName name, Optional<Window> window, Optional<Expression> filter, boolean distinct, List<Expression> arguments)
+    public FunctionCall(NodeLocation location, QualifiedName name, Optional<Window> window, Optional<Expression> filter, Optional<List<SortItem>> sortItems, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.of(location), name, window, filter, distinct, arguments);
+        this(Optional.of(location), name, window, filter, sortItems, distinct, arguments);
     }
 
-    private FunctionCall(Optional<NodeLocation> location, QualifiedName name, Optional<Window> window, Optional<Expression> filter, boolean distinct, List<Expression> arguments)
+    private FunctionCall(Optional<NodeLocation> location, QualifiedName name, Optional<Window> window, Optional<Expression> filter, Optional<List<SortItem>> sortItems, boolean distinct, List<Expression> arguments)
     {
         super(location);
         requireNonNull(name, "name is null");
         requireNonNull(window, "window is null");
         requireNonNull(filter, "filter is null");
         requireNonNull(arguments, "arguments is null");
+        requireNonNull(sortItems, "sortItems is null");
 
         this.name = name;
         this.window = window;
         this.distinct = distinct;
         this.arguments = arguments;
         this.filter = filter;
+        this.sortItems = sortItems;
```

Comment:
I should not happen, but I don't think it would cause problems. But I can validate.